.github/workflows/pull_request.yml

```diff
@@ -550,6 +550,13 @@ jobs:
         with:
           clear-repository: true
           submodules: true
+      - name: Apply sparse checkout for contrib # in order to check that it doesn't break build
```

Comment:
Please, don't forget to roll it back. Unfortunately, it doesn't increase the checkout speed, only the allocated disk space.

Reply:
I had no intention of rolling it back, I'm going to keep it. See the comment below

Reply:
> It is necessary for at least one build. Because we must check that it works. Imagine that some contributor cloned the repo with "sparse checkout" following the documentation. And another contributor has full contribs. The second contributor adds some code that uses a new boost module, for example. It compiles on their machine. It compiles in the CI.

I see. In this case, the job is done very well. We have all submodules cloned in the `checkout` step before, and the following test is done faster.

Maybe, if we do it in `BuilderDebDebug`, it makes sense to adjust every workflow having it?

- .github/workflows/backport_branches.yml
- .github/workflows/master.yml
- .github/workflows/release_branches.yml

Reply:
> Maybe, if we do it in BuilderDebDebug, it makes sense to adjust every workflow having it?

I think that having it in `pull_request.yml` only is enough to avoid breaking the build with "sparse checkout". But I don't have a strong opinion on that, we can add it to all workflows for consistency.

Also it makes sense to add an option to use `update-submodules.sh` in https://github.com/ClickHouse/checkout (and avoid the ugly trick with `rm -rf`), but it will make the action less universal

Reply:
@Felixoid, I added the same change to other workflows

P.S. It would be great to refactor these yamls somehow and reduce the amount of copy-paste